drivers/wifi/wfx200/wfx200_internal.h

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2021 grandcentrix GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __ZEPHYR_DRIVERS_WIFI_SILABS_WFX200_INTERNAL_H_
+#define __ZEPHYR_DRIVERS_WIFI_SILABS_WFX200_INTERNAL_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <net/ethernet.h>
+#include <net/wifi_mgmt.h>
+
+#include <sl_wfx_constants.h>
+
+enum wfx200_event {
+	WFX200_CONNECT_EVENT = 0,
+	WFX200_CONNECT_FAILED_EVENT,
+	WFX200_DISCONNECT_EVENT,
+	WFX200_AP_START_EVENT,
+	WFX200_AP_START_FAILED_EVENT,
+	WFX200_AP_STOP_EVENT,
+};
+
+struct wfx200_queue_event {
+	enum wfx200_event ev;
+	union {
+
+	};
+};
+
+struct wfx200_gpio {
+	const struct device *dev;
+	uint8_t pin;
+};
+
+enum wfx200_state {
+	WFX200_STATE_IDLE = 0,
+	WFX200_STATE_INITIALIZED,
+	WFX200_STATE_INTERFACE_INITIALIZED,
+	WFX200_STATE_AP_MODE,
+	WFX200_STATE_STA_MODE,
+};
+
+struct wfx200_dev {
+	struct net_if *iface;
+	const struct device *dev;
+
+	const struct device *spi;
```

Comment:
New driver should probably use spi_dt_spec and gpio_dt_spec, see e.g. https://github.com/zephyrproject-rtos/zephyr/pull/37498